tools/test/examples/examples_lib.py

```diff
@@ -380,31 +382,60 @@ def compile_repos(config, toolchains, targets, profile, verbose, examples):
                                                                 valid_choices(example['toolchains'], toolchains),
                                                                 example['features']):
-                    print("Compiling %s for %s, %s" % (name, target, toolchain))
+                    
                     build_command = ["mbed-cli", "compile", "-t", toolchain, "-m", target] + (['-v'] if verbose else [])
-
                     if profile:
                         build_command.append("--profile")
                         build_command.append(profile)
-
-                    proc = subprocess.Popen(build_command)
-
-                    proc.wait()
+                    
+                    print("Compiling [%s] for [%s] with toolchain [%s]\n\n>  %s" % (name, target, toolchain, " ".join(build_command)))
+                    
+                    proc = subprocess.Popen(build_command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+                    std_out, std_err = proc.communicate()
+                    print ("\n#### STDOUT ####\n%s\n#### STDERR ####\n%s\n#### End of STDOUT/STDERR ####\n" % (std_out,std_err))
+                    
+                    if test_example:
+                        log = example['compare_log'].pop(0)
+                        image = fetch_output_image(std_out)
+                        if image:
+                            image_info = [{"binary_type": "bootable","path": normpath(join(name,image)),"compare_log":log}]
+                        else:
+                            print ("Warning: could not found built image for example %s" % name)
+                        
                     example_summary = "{} {} {}".format(name, target, toolchain)
                     if proc.returncode:
                         failures.append(example_summary)
                     else:
                         successes.append(example_summary)
+                        if test_example:
+                            test_group = "{}-{}-{}".format(target, toolchain, example['baud_rate'])
+                            if example['test'] and image:
+                                if not test_json['builds'].has_key(test_group):
+                                    test_json['builds'][test_group] = {
+                                        "platform":target ,
+                                        "toolchain": toolchain ,
+                                        "base_path": os.getcwd() ,
+                                        "baud_rate": int(example['baud_rate']), 
+                                        "tests":{} }
+                                test_json['builds'][test_group]['tests'][name]={"binaries":image_info}
+                        else:
+                            print("Warning: Test for %s will not be generated." % name)
+                            print("One or more of 'test' 'baud_rate' and 'compare_log' keys are missing from the json file\n")
```

Comment:
nit pick on wording, missing some commas:
```suggestion
                            print("One or more of 'test', 'baud_rate', and 'compare_log' keys are missing from the example config json file\n")
```